client/remote_test.go

```diff
@@ -140,7 +134,16 @@ func TestMain(m *testing.M) {
 		log.Fatal(err)
 	}
 
-	os.Exit(m.Run())
+	ret := m.Run()
+	// clean up server listeners
+	if s.UnixListener != nil {
```

Comment:
Does the [`Serve`](https://github.com/cloudflare/gokeyless/pull/130/files#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bL336) function not clean these up properly?